src/state/utils.ts

```diff
@@ -6,101 +6,117 @@ import { AxiosResponse } from 'axios';
-                          min: element.min as number,
-                          max: element.max as string,
-                          valueSet: findValueSet.resource.concept
-                        });
-                    }
-                  } else {
-                    element.path &&
-                      type.code &&
-                      element.definition &&
-                      attributes.push({
-                        definition: element.definition,
-                        path: element.path,
-                        type: type.code,
-                        min: element.min as number,
-                        max: element.max as string
-                      });
+        element.type.forEach((types: IElementDefinitionType) => {
+          if (element.path && types.code && element.definition) {
+            if (types.code === 'code') {
+              const valueSet: any[] = [];
+              element.binding?.extension?.forEach((extension) => {
+                if (extension.valueString) {
+                  const findValueSet = valueSetRequest?.data.entry.find(
+                    (value: FetchedDataCodeSystem) =>
+                      value.resource.name === extension.valueString
+                  );
+                  if (findValueSet) {
+                    findValueSet.resource.concept.forEach((vs: any) => {
+                      valueSet.push(vs);
+                    });
                   }
-                });
-              } else {
-                attributes.push({
-                  path: element.path,
-                  type: type.code,
-                  definition: element.definition,
-                  min: element.min as number,
-                  max: element.max as string
-                });
-              }
+                }
+              });
+              const newAttribute = {
+                definition: element.definition,
+                path: element.path,
+                type: types.code,
+                min: element.min as number,
+                max: element.max as string,
+                binding: {
+                  valueSet: valueSet.length > 0 ? valueSet : undefined,
+                  strength: element.binding?.strength
+                }
+              };
+              attributes.push(newAttribute);
+            } else {
+              attributes.push({
+                path: element.path,
+                type: types.code,
+                definition: element.definition,
+                min: element.min as number,
+                max: element.max as string
+              });
             }
-          });
-        }
+          }
+        });
       }
-    });
+    }
   });
   return attributes;
 };
 
+/**
+ * Create simplified attributes from axios response in middleware
+ * @param data axios response for data
+ * @param valueSet value set response
+ */
+export const createSimplifiedAttributes = (
+  data: AxiosResponse<any>,
+  valueSet: AxiosResponse<any>
```

Comment:
I think those 2 arguments can be better typed than AxiosResponse<any>

createSimplifiedAttribudes is called 2 times with a response from a StructureDef request for 1st argument and a response from a CodeSystem request for 2nd argument.

This information should be enough to type this like so :

```suggestion
  data: AxiosResponse<RELATED_TYPE_TO_IStructureDefinition>,
  valueSet: AxiosResponse<RELATED_TYPE_TO_ICodeSystem>
```

Or, better, you could directly declare your functions arguments like so :

```suggestion
  data: IStructureDefinition,
  valueSet: ICodeSystem
```